controllers/scenario/scenario_adapter.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2022.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scenario
+
+import (
+	"context"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/go-logr/logr"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/api/v1alpha1"
+	"github.com/redhat-appstudio/integration-service/gitops"
+	"github.com/redhat-appstudio/operator-goodies/reconciler"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Adapter holds the objects needed to reconcile a Release.
+type Adapter struct {
+	application *applicationapiv1alpha1.Application
+	scenario    *v1alpha1.IntegrationTestScenario
+	logger      logr.Logger
+	client      client.Client
+	context     context.Context
+}
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(application *applicationapiv1alpha1.Application, scenario *v1alpha1.IntegrationTestScenario, logger logr.Logger, client client.Client,
+	context context.Context) *Adapter {
+	return &Adapter{
+		application: application,
+		scenario:    scenario,
+		logger:      logger,
+		client:      client,
+		context:     context,
+	}
+}
+
+// EnsureCreatedScenarioIsValid is an operation that ensures created IntegrationTestScenario is valid
+// in case it is, set its owner reference
+func (a *Adapter) EnsureCreatedScenarioIsValid() (reconciler.OperationResult, error) {
+	if meta.IsStatusConditionTrue(a.scenario.Status.Conditions, gitops.HACBSIntegrationStatusCondition) {
```

Comment:
I think we first want to check if the `a.scenario` is valid by checking if `a.application` is nil. If it's not nil we should then check if the owner reference is set - if it's not set already, we should try to set the owner reference.
If any of the actions from above fail, we should then check if `a.scenario` has already been marked as invalid (with this check) - if it hasn't been already marked as invalid, only then the update should be made to the scenario status.

Let say the user puts the wrong application name into the ITS. We then mark it as invalid in this function. Once they realize that that they put the wrong application name, they try to fix it. We should then re-check the ITS and either leave the invalid status (by **not** updating the ITS resource, otherwise we'd end up in a loop) or change the status to valid (true).